File: 2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        teams = []
        chem = 0
        skill.sort()
        i, j = 0, len(skill) - 1
        number = skill[i] + skill[j]
        while i < j:
            if(skill[i] + skill[j] == number):
                teams.append((skill[i], skill[j]))
                i += 1
                j -= 1
            else:
                return -1

        for (x, y) in teams:
            chem += x*y
        
        return chem

File: 2581-divide-players-into-teams-of-equal-skill/divide-players-into-teams-of-equal-skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        sort(skill.begin(), skill.end());
        long long chem = 0;
        int length = skill.size();
        int teamSum = skill[0] + skill[length - 1];
        for(int i = 0; i < length/2; i++){
            if(skill[i] + skill[length - i - 1] == teamSum){
                chem += skill[length - i - 1] * skill[i] ;
            }
            else{
                return -1;
            }
        }
        return chem;
    }
};